src/common/com/intellij/plugins/haxe/ide/module/HaxeModuleBuilder.java

```diff
@@ -51,6 +74,75 @@ public void moduleCreated(@NotNull Module module) {
     final CompilerModuleExtension model = (CompilerModuleExtension)CompilerModuleExtension.getInstance(module).getModifiableModel(true);
     model.setCompilerOutputPath(model.getCompilerOutputUrl());
     model.inheritCompilerOutputPath(false);
+
+    final Project project = module.getProject();
+    List<Pair<String, String>> sourcePaths = getSourcePaths();
+    String srcPath = null;
+
+    if ((sourcePaths.size() > 0)) {
+      srcPath = sourcePaths.get(0).getFirst();
+      VirtualFile dir = VirtualFileManager.getInstance().findFileByUrl(VfsUtilCore.pathToUrl(srcPath));
+
+      if (dir != null) {
+        try {
+          VirtualFile mainClassFile = dir.findFileByRelativePath("Main.hx");
+          if (mainClassFile == null) {
+            HaxeFileTemplateUtil
+              .createClass("Main", "", PsiManager.getInstance(project).findDirectory(dir), "HaxeMainClass",
+                           null);
+          }
+        }
+        catch (Exception e) {
```

Comment:
Are you sure you want to catch /every/ exception?  I'd much prefer catching only the exception that the code can throw.  Also, if we just continue after the exception, then we are changing settings to work with a non-existent class. The best course of action is to catch recoverable errors and let the rest fly past.